pkg/v1/workflow/declaration.go

```diff
@@ -237,7 +237,7 @@ func (w *workflowDeclarationImpl[I, O]) Task(opts create.WorkflowTask[I, O], fn
 	genericFn := func(ctx worker.HatchetContext, input I) (*any, error) {
 		// Use reflection to call the specific function
 		fnValue := reflect.ValueOf(fn)
-		inputs := []reflect.Value{reflect.ValueOf(input), reflect.ValueOf(ctx)}
+		inputs := []reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(input)}
```

Comment:
@grutt would be very helpful to double check that these are all correct. I first noticed a panic on an on-failure step, but I suspect all of these are in the wrong order?

Reply:
![image](https://github.com/user-attachments/assets/a5cffd48-9fa5-4e75-988d-bd9054876ea5)
good catch, i think the rest are ok -- will look closer tomorrow